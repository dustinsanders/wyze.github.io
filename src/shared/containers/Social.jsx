// @flow

import type { Profile, Profiles, Style, VNode } from '../../types'
import { css, media } from 'glamor'
import { small } from '../styles/breakpoints'
import Box from '../components/Box'
import Link from '../components/Link'
import Section from '../components/Section'
import preact from 'preact'

type Props = {
  profiles: Profiles,
}

const link: Style = css({
  display: 'inline-block',
  height: '5.7em',
  minWidth: '5.7em',
})

const social: Style = css(
  {
    flex: '0 0 50%',
    paddingTop: '.4em',
    margin: '.5em 0',
  },
  media(small, {
    flexBasis: '25%',
    margin: 0,
  }),
)

const Social = ({ profiles }: Props): VNode => (
  <Box wrap title="Me Around The Internet">
    {profiles.map(({ component: Component, to }: Profile): VNode => (
      <Section key={to} styles={social}>
        <Link styles={link} to={to}>
          <Component />
        </Link>
      </Section>
    ))}
  </Box>
)

export default Social
