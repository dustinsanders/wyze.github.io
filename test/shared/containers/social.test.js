// @flow

import Social from '../../../src/shared/containers/Social'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const profiles = [
  // eslint-disable-next-line react/jsx-filename-extension
  { component: <svg />, to: '//abc.xyz' },
]

describe('<Social />', () => {
  it('renders', snapshot.bind(null, Social, { profiles }))
})
