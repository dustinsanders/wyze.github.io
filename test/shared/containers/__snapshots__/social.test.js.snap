// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Social /> renders 1`] = `
"<Box wrap title=\\"Me Around The Internet\\">
  <Section styles=\\"css-f3ixb5\\">
    <Link styles=\\"css-5j31gr\\" to=\\"//abc.xyz\\">
      <[object Object]></[object Object]>
    </Link>
  </Section>
</Box>"
`;
